contracts/Colony.sol

```diff
@@ -125,11 +127,36 @@ contract Colony is ColonyStorage {
   function getDomainCount() public view returns (uint256) {
     return domainCount;
   }
-
   function setFunctionReviewers(bytes4 _sig, uint8 _firstReviewer, uint8 _secondReviewer)
   private
   {
     uint8[2] memory _reviewers = [_firstReviewer, _secondReviewer];
     reviewers[_sig] = _reviewers;
   }
+
+  modifier verifyKey(bytes key) {
+    uint256 colonyAddress;
+    uint256 skillid;
+    uint256 userAddress;
+    assembly {
+        colonyAddress := mload(add(key,32))
```

Comment:
Just to confirm I understand this. We start from the 32 bit onwards because that first one contains the bytes length value?

Reply:
32nd *byte* onwards, but yes. The first slot (32 bytes) of a dynamic array contains its length.